plugin/src/ios/appDelegate.ts

```diff
@@ -0,0 +1,98 @@
+import { ConfigPlugin, withAppDelegate } from "@expo/config-plugins";
+import { mergeContents } from "@expo/config-plugins/build/utils/generateCode";
+export const withIosAppDelegate: ConfigPlugin = (config) => {
+  return withAppDelegate(config, (cfg) => {
+    const { modResults } = cfg;
+
+    // method to invoke voip registration
+    // I decided to use this as soon as the app starts to avoid js delay issues
+    const methodInvocationBlock = `[RNVoipPushNotificationManager voipRegistration];`;
+
+    // https://regex101.com/r/mPgaq6/1
+    const methodInvocationLineMatcher =
+      /(?:self\.moduleName\s*=\s*@\"([^"]*)\";)|(?:(self\.|_)(\w+)\s?=\s?\[\[UMModuleRegistryAdapter alloc\])|(?:RCTBridge\s?\*\s?(\w+)\s?=\s?\[(\[RCTBridge alloc\]|self\.reactDelegate))/g;
+
+    // https://regex101.com/r/nHrTa9/1/
+    // if the above regex fails, we can use this one as a fallback:
+    const fallbackInvocationLineMatcher =
+      /-\s*\(BOOL\)\s*application:\s*\(UIApplication\s*\*\s*\)\s*\w+\s+didFinishLaunchingWithOptions:/g;
+
+    if (!modResults.contents.includes("#import <PushKit/PushKit.h>")) {
+      modResults.contents = modResults.contents.replace(
+        /#import "AppDelegate.h"/g,
+        `#import "AppDelegate.h"
+#import <PushKit/PushKit.h>
```

Comment:
as a suggestion, there is a function in expo plugin to handle imports.

`const { addObjcImports } = require('@expo/config-plugins/build/ios/codeMod');`

Then, you can use it like this.

`
modResults.contents = addObjcImports(modResults.contents, ['#import <PushKit/PushKit.h>', '#import "RNVoipPushNotificationManager.h"', '#import "RNCallKeep.h"'])
`

Reply:
You can see a full example in this project.

https://github.com/GetStream/stream-video-js/blob/main/packages/react-native-sdk/expo-config-plugin/src/withPushAppDelegate.ts

Reply:
This should be a comment, not a review.

Reply:
Thanks @AZReed. Kinda confuse about this. Should I proceed making this changes you suggest?

Reply:
That's the problem with reviews. It's hard to tell of someone is just flexing or not. They say "suggestion" but open a review.

Reply:
Hey @jexodusmercado 
Maybe it should have been a comment, not sure, my apologies for the confusion. I press the `comment` option in the review, not the `request changes`, I though that was clear enough.

It's your time and you already did enough, so it's up to you.

Thanks for your work by the way, it's awesome!